lib/signaling/v2/twilioconnectiontransport.js

```diff
@@ -340,7 +385,27 @@ function setupEventListeners(transport) {
       transport.disconnect();
       return;
     }
-    transport.disconnect(new SignalingConnectionError());
+    if (transport._reconnectAttemptsLeft <= 0) {
+      transport.disconnect(new SignalingConnectionError());
+    }
+    reconnect();
+  }
+
+  function reconnect() {
+    transport._reconnectAttemptsLeft--;
+    const { maxReconnectAttempts } = transport._options;
+    const reconnectAttempts = maxReconnectAttempts - transport._reconnectAttemptsLeft;
+    const backOffMs = (1 << reconnectAttempts) * transport._reconnectBackOffMs;
```

Comment:
Nitpick: I think `2 ** x` is more readable than `1 << x`, e.g.

```js
(2 ** reconnectAttempts) * transport._reconnectionBackOffMs;
```

Reply:
@markandrus I think (1 << x) is pretty well established as power-of-2. So I would like to keep this as is.